internal/config/config.go

```diff
@@ -153,6 +135,14 @@ func (c *AuthConfig) HasEnvToken() bool {
 			return true
 		}
 	}
+	// TODO: This is _extremely_ knowledgable about the implementation of TokenFromEnvOrConfig
```

Comment:
Calling this out for attention.

Reply:
Agreed. We should create a function `HasEnvToken` in `go-gh` that abstracts away this knowledge.

Reply:
I'm going to punt that from this PR since it's a bigger change and this is already pretty hefty.